### PR TITLE
Add a trunk-build github action

### PR DESCRIPTION
**Pull request by NathanReb**
Adds an action that builds with the latest supported trunk. We'll need this for incremental trunk support and I'd like to see it run on #487.

I heavily reused the work of @shym from #410, thank you for this contribution!

### CODE REVIEW

**Comment by NathanReb** [comment]
The trunk build fails atm of course because our opam bounds don't allow for OCaml 5.3.

File: .github/workflows/trunk-build.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Build on `trunk`
+
+on:
+  push:
```
Comment (patricoferris) [inline .github/workflows/trunk-build.yml:4]: Do we want this to also run on just _any_ push to _any_ branch? Or only on pushes to `main` and pulls?

**Review by patricoferris — APPROVED** [review]
This looks good :))

File: .github/workflows/trunk-build.yml (hunk 1 above)
Comment (NathanReb) [reply .github/workflows/trunk-build.yml:4]: I'd say only on push events to `main`, I'll update the action!

Comment (NathanReb) [reply .github/workflows/trunk-build.yml:4]: For PRs I'd keep any branch though, does that sound good?

**Pull request merged**